ui-kit/src/main/kotlin/org/p2p/uikit/atoms/UiKitBadgeView.kt

```diff
@@ -0,0 +1,91 @@
+package org.p2p.uikit.atoms
+
+import android.content.Context
+import android.graphics.Paint
+import android.graphics.drawable.ColorDrawable
+import android.graphics.drawable.LayerDrawable
+import android.graphics.drawable.ShapeDrawable
+import android.graphics.drawable.shapes.RoundRectShape
+import android.util.AttributeSet
+import android.view.Gravity
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.core.content.ContextCompat
+import org.p2p.uikit.R
+import org.p2p.uikit.utils.toPx
+
+private const val RADIUS_LIST_SIZE = 8
+private const val VERTICAL_PADDING_DP = 0
+private const val HORIZONTAL_PADDING_DP = 8
+
+class UiKitBadgeView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : AppCompatTextView(context, attrs, defStyleAttr) {
+
+    enum class Shape {
+        OVAL,
+        RECTANGLE
+    }
+
+    init {
+        context.obtainStyledAttributes(attrs, R.styleable.UiKitBadgeView).use { typedArray ->
+            val defaultTextColor = ContextCompat.getColor(context, R.color.mountain)
```

Comment:
Please use native method `context.getColor`
It uses compat for lower OS versions and standard get color for newer versions automatically

Reply:
You're right, `context.getColor()` is preferred in our case. But you're a bit wrong about method, cause `context.getColor()` doesn't make any compatibility checks and `ContextCompat.getColor()` does. But it useless for us, cause it uses native method for 23+ SDK versions and we have 26 as minimum anyway